ospd_openvas/openvas_db.py

```diff
@@ -60,219 +44,234 @@
+    from a KB to another."""
+
+    def __init__(self):
+        # Path to the Redis socket.
+        self.db_address = ""
+
+        # Name of the namespace usage bitmap in redis.
+        self.dbindex_name = "GVM.__GlobalDBIndex"
+        self.max_dbindex = 0
+        self.db_index = 0
+        self.rediscontext = None
+
+    def get_db_connection(self):
+        """ Retrieve the db address from openvassd config.
+        """
+        result = subprocess.check_output(
+            ['openvassd', '-s'], stderr=subprocess.STDOUT)
+
+        path = None
+        if result:
+            result = result.decode('ascii')
+            for conf in result.split('\n'):
+                if conf.find('db_address') == 0:
+                    path = conf.split('=')
+                    break
+
+        if not path:
+            raise OSPDOpenvasError('Not possible to find the path to '
+                               'the redis socket.')
+
+        self.db_address = str.strip(path[1])
+
+
+    def max_db_index(self):
+        """Set the number of databases have been configured into kbr struct.
+        """
+        ctx = self.kb_connect()
+        resp = ctx.config_get('databases')
+
+        if len(resp) == 1:
+            self.max_dbindex = int(resp.get('databases'))
+        else:
+            raise OSPDOpenvasError('Redis Error: Not possible '
+                         'to get max_dbindex.')
+
+    def set_redisctx(self, ctx):
+        """ Set the current rediscontext.
+        """
+        self.rediscontext = ctx
+
+    def db_init(self):
+        """ Set db_address and max_db_index. """
+        self.get_db_connection()
+        self.max_db_index()
+
+    def try_database_index(self, ctx, i):
+        """ Check if it is already in use. If not set it as in use and return.
+        """
+        try:
+            resp = ctx.hsetnx(self.dbindex_name, i, 1)
+        except:
+            return 2
+
+        if isinstance(resp, int) is False:
+            return 2
+
+        if resp == 1:
+            return 1
+
+    def kb_connect(self, dbnum=0):
+        """ Connect to redis to the given database or to the default db 0 .
+        Arguments:
+            dbnum (int): The db number to connect to.
+
+        Return a redis context on success or 2 on error
+        """
+        self.get_db_connection()
+
+        try:
+            ctx = redis.Redis(unix_socket_path=self.db_address,
+                              db=dbnum,
+                              socket_timeout=SOCKET_TIMEOUT, charset="latin-1",
+                              decode_responses=True)
+        except ConnectionError as e:
+            return {"error": str(e)}
+        self.db_index = dbnum
+        return ctx
+
+    def db_find(self, patt):
+        """ Search a pattern inside all kbs. When find it return it.
+        """
+        for i in range(0, self.max_dbindex):
+            ctx = self.kb_connect (i)
+            if ctx.keys(patt):
                 return ctx
```

Comment:
Please add a return value for the not found case. See [pep8 "Be consistent in return statements"](https://www.python.org/dev/peps/pep-0008/#programming-recommendations)